python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -194,26 +194,25 @@ def initialize_progress_bars(self, index: int, verbose_progress: bool) -> int:
         """Create progress bars at the given index (line offset in console).
 
         For AllToAllOperator, zero or more sub progress bar would be created.
-        Return the number of progress bars created for this operator.
+        Return the number of enabled progress bars created for this operator.
         """
         is_all_to_all = isinstance(self.op, AllToAllOperator)
         # Only show 1:1 ops when in verbose progress mode.
-        enabled = is_all_to_all or (
-            DataContext.get_current().enable_progress_bars and verbose_progress
+        progress_bar_enabled = DataContext.get_current().enable_progress_bars and (
+            is_all_to_all or verbose_progress
         )
         self.progress_bar = ProgressBar(
             "- " + self.op.name,
             self.op.num_outputs_total(),
             index,
-            enabled=enabled,
+            enabled=progress_bar_enabled,
         )
-        if enabled:
-            num_bars = 1
-            if is_all_to_all:
-                num_bars += self.op.initialize_sub_progress_bars(index + 1)
-        else:
-            num_bars = 0
-        return num_bars
+        num_progress_bars = 1
+        if is_all_to_all:
+            # Initialize must be called for sub progress bars, even the
+            # bars are not enabled via the DataContext.
+            num_progress_bars += self.op.initialize_sub_progress_bars(index + 1)
+        return num_progress_bars if progress_bar_enabled else 0
```

Comment:
Original code wants to return number of enabled progress bars, although I think that number of created "ProgressBar" instances would be more meaningful.

I preserved original behaviour since I did not follow the calls if the visibility of the progress bar is really what client wants to know.

Reply:
yes, let's keep existing behavior.